pkg/plugins/manager/installer.go

```diff
@@ -156,9 +130,61 @@ func (m *PluginInstaller) install(ctx context.Context, pluginID, version string,
 		return nil, err
 	}
 
+	if extractedArchive.ID != pluginID {
+		return nil, fmt.Errorf("plugin ID mismatch: expected %s, got %s", pluginID, extractedArchive.ID)
```

Comment:
Returning an error isn't enough here unless we are fine to keep the bad archive on disk FYI

Reply:
yeah, I was debating about this one. Since we didn't have this check before I wanted to keep the previous behavior. Maybe it's better to just log this as a warning? Note that in the case of upgrade, we have already deleted the previous version at this point so if we also remove this new "wrong" version, we will be effectively uninstalling the plugin, rather than upgrading. Thoughts?

Reply:
Yeah the behaviour is pretty weird since if they try to install a "wrong" version, then it won't load. But if they restart it will load fine. In that case I think it's probably best to warn so that it will load first time.

Maybe let's create an issue to figure out if we can improve this so that you
- Only get what you request (based on plugin id + version)
- Won't remove an existing installation unless we know we have a viable upgrade